### PR TITLE
Add dynamic state description provider

### PR DESCRIPTION
**Pull request by cdjackson**
This adds a ```DynamicStateDescriptionProvider``` to provide state descriptions for channels if required.
Signed-off-by: Chris Jackson <chris@cd-jackson.com>

### CODE REVIEW

File: org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java [hunk 1]
```diff
@@ -457,6 +472,17 @@ public void run() {
...
     }
 
+    @Override
+    public @Nullable StateDescription getStateDescription(Channel channel,
+            @Nullable StateDescription originalStateDescription, @Nullable Locale locale) {
+        if (stateDescriptions.containsKey(channel.getUID())) {
+            logger.trace("Returning new stateDescription for {}", channel.getUID());
+            return stateDescriptions.get(channel.getUID());
+        } else {
+            return originalStateDescription;
```
Comment (hsudbrock) [inline org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java]: From the JavaDoc of the `DynamicStateDescriptionProvider`, I take that this method should return `null` (and not the `originalStateDescription`) if no dynamic state description shall be provided. (Citation from JavaDoc: `@return state description or null if none provided`)

```suggestion
            return null;
```

**Comment by cdjackson** [comment]
Thanks Henning - I've updated as suggested.

Your English is too good, and mine is too lazy! ;)

**Comment by hsudbrock** [comment]
Jenkins is taking some time to finish... If it is stuck, could you trigger another run @cdjackson ? (I cannot, AFAIK.)

**Comment by kaikreuzer** [comment]
It has finished successfully: https://openhab.ci.cloudbees.com/view/Pull%20Request%20Builds/job/PR-org.openhab.binding.zigbee/277/
I guess it is Github, which still has issues...

**Comment by cdjackson** [comment]
I normally just close/reopen the PR which re-triggers CI.

I’m not sure if this is still using cloudbees (I think so) but Kai is moving to a new server at ci.openhab.org.

**Pull request merged**